web/src/composebox_typeahead.ts

```diff
@@ -1180,6 +1185,31 @@ export function content_typeahead_selected(
     return beginning + rest;
 }
 
+function will_produce_broken_stream_topic_link(topic_name: string): boolean {
+    return /(\*+)|(.*`.*`.*)/.test(topic_name);
```

Comment:
The links for topic names with double backticks are now generated correctly, but the double backticks cause the enclosed part of the topic name to render as an inline code block. 
I don't know what to do in this case since backslash escaping is disabled.

Reply:
Yeah, I think there's no great answer for that at present. This is #18202.

As long as the link works correctly, though, that's a much less severe issue than #19873 is.

A possible workaround is to use HTML character references, as suggested here: https://github.com/zulip/zulip/issues/18202#issuecomment-1900927189
I'd definitely leave that as a separate later commit, though.